livekit-android-sdk/src/main/java/io/livekit/android/room/track/video/CameraCapturerUtils.kt

```diff
@@ -210,29 +221,43 @@ object CameraCapturerUtils {
         return targetDeviceName
     }
 
-    /**
-     * Finds the device id of a camera that matches the [predicate].
-     */
-    fun CameraEnumerator.findCamera(predicate: (deviceName: String) -> Boolean): String? {
-        for (deviceName in deviceNames) {
-            if (predicate(deviceName)) {
-                return deviceName
-            }
-        }
-        return null
-    }
+    data class CameraDeviceInfo(val deviceId: String, val position: CameraPosition?, val physicalId: String?)
 
     /**
-     * Returns the camera position of a camera, or null if neither front or back facing (e.g. external camera).
+     * Returns information about a camera by searching for the specified device ID.
+     * If the device ID matches a logical camera ID, returns that camera's info.
+     * If the device ID matches a physical camera ID on Android P and above, returns
+     * the logical camera info with the physical ID relationship.
+     *
+     * @param cameraManager The system camera manager
+     * @param predicate
+     * @return PhysicalCameraInfo with camera details or null if not found
      */
-    fun CameraEnumerator.getCameraPosition(deviceName: String?): CameraPosition? {
-        if (deviceName == null) {
-            return null
-        }
-        if (isBackFacing(deviceName)) {
-            return CameraPosition.BACK
-        } else if (isFrontFacing(deviceName)) {
-            return CameraPosition.FRONT
+    fun CameraEnumerator.findCamera(
+        cameraManager: CameraManager,
+        predicate: (deviceId: String, position: CameraPosition?) -> Boolean,
+    ): CameraDeviceInfo? {
+        for (id in cameraManager.cameraIdList) {
+            val ch = cameraManager.getCameraCharacteristics(id)
+            val lensFacing = ch.get(CameraCharacteristics.LENS_FACING)
+            val position = when (lensFacing) {
+                CameraCharacteristics.LENS_FACING_FRONT -> CameraPosition.FRONT
+                CameraCharacteristics.LENS_FACING_BACK -> CameraPosition.BACK
+                else -> null
+            }
+            // First check if deviceId is a direct logical camera ID
+            if (predicate(id, position)) return CameraDeviceInfo(id, position, null)
+
+            // Then check if deviceId is a physical camera ID in a logical camera
+            if (VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                val ch2 = cameraManager.getCameraCharacteristics(id)
+
+                for (physicalId in ch2.physicalCameraIds) {
+                    if (predicate(physicalId, position)) {
+                        return CameraDeviceInfo(id, position, physicalId)
+                    }
+                }
+            }
```

Comment:
New revised `findCamera` function that supports passing physical id and returns an object with position.